### DISCUSSION TITLE
rustdesk public server for accessibility-test?

### DISCUSSION THREAD

**Discussion by Michael1972** [rustdesk/rustdesk#9602] [Q&A] [answered]
Hi to the community,

I'm working voluntarily for a blind- and visually-impaired-organisation in Germany. We would like to provide a remote-server, and rustdesk could be a good choice. Before setting up our own server, is it possible to test rustdesk in terms of accessibility publicly? I couldn't find out how to access the public server at rustdesk.cyberrange.rit.edu, and because we are all working voluntarily, it would be good to know, if rustdesk's webinterface and the remote-sessions are accessible with screenreaders.

Thanks in advance for any help and best

Michael

**Comment by basilgello** [comment]
Hi @Michael1972, it is a very noble initiative!

I am also a volunteer but let me clearly understand what you currently have in terms of infrastructure.

You want to deploy your own publicly accessible Rustdesk ID and Relay server on RIT infra?
But before that, you want to check if Rustdesk client app follows some accessibility guidelines?

Or, you want to deploy server first just to run your accessibility tests withput interference with Rustdesk public infrastructure?

**Comment by Michael1972** [comment, marked as answer]
Hi Basil,

thanks for your answer. I want to set up my own self-hosted rustdesk-server in an own environment. My organisation has some servers, currently used for different services by our blind users. Of course, setting up the rustdesk-instance takes some time. I thought, I could have a look at a publicly available rustdesk-instance, in order to get an impression of its accessibility. Checking, if rustdesk is accessible with screenreaders and if so, if the amount of accessibility fits our needs, would take round about 30 minutes. Setting up and deploying an own rustdesk-server would take much more time. And I thought I could get an impression of rustdesk on a public instance, before spending the work and time, setting up an own server and possibly noticing, that it is inaccessible.

**Comment by xlionjuan** [comment]
For Web client, they are improving it (I mean functionality, not accessibility) recently, but it have a serious problem to solve, encryption.

Web client don't support encryption because the server don't support WebSocket with TLS (`wss://`), and you can't provide Web client with HTTPS, because browsers will prohibit you to connect you to `ws:\\` if you're using HTTPS, you can only allowed to connect to `ws:\\` if you're using HTTP.

There is a workaround, use Traefik to convert `ws:\\` to `wss:\\`

https://github.com/rustdesk/rustdesk/discussions/6023
